jupyterhub/apihandlers/auth.py

```diff
@@ -31,33 +33,43 @@ def get(self, token):
             self.db.delete(orm_token)
             self.db.commit()
             raise web.HTTPError(404)
+        # record activity whenever we see a token
+        orm_token.last_activity = datetime.utcnow()
+        self.db.commit()
         self.write(json.dumps(model))
 
     @gen.coroutine
     def post(self):
-        user = self.get_current_user()
+        requester = user = self.get_current_user()
         if user is None:
             # allow requesting a token with username and password
             # for authenticators where that's possible
             data = self.get_json_body()
             try:
-                user = yield self.login_user(data)
+                requester = user = yield self.login_user(data)
             except Exception as e:
                 self.log.error("Failure trying to authenticate with form data: %s" % e)
                 user = None
             if user is None:
                 raise web.HTTPError(403)
         else:
             data = self.get_json_body()
-            # admin users can request 
-            if data and data.get('username') != user.name:
-                if user.admin:
-                    user = self.find_user(data['username'])
-                    if user is None:
-                        raise web.HTTPError(400, "No such user '%s'" % data['username'])
-                else:
+            # admin users can request tokens for other users
```

Comment:
can an admin user request a token for another admin user? I assume that is okay.

Reply:
Yes, any other user.